machida3/machida.pony

```diff
@@ -188,7 +188,9 @@ class PyPartitionFunction
         Fail()
       end
 
-      let ret = String.copy_cstring(@py_bytes_or_unicode_as_char(ps))
+      let py_string_size = @py_bytes_or_unicode_size(ps)
+
+      let ret = String.copy_cpointer(@py_bytes_or_unicode_as_char(ps), py_string_size)
```

Comment:
Should this use the `Machida.py_bytes_or_unicode_to_pony_string` primitive?

Reply:
Yeah, I can fix this one.